src/GeniusYield/Types/TxMetadata.hs

```diff
@@ -0,0 +1,114 @@
+module GeniusYield.Types.TxMetadata (
+  GYTxMetadata (..),
+  GYTxMetadataValue,
+  gyTxMetaMap,
+  gyTxMetaList,
+  gyTxMetaNumber,
+  gyTxMetaBytes,
+  gyTxMetaText,
+  gyMetaTextChunks,
+  gyMetaBytesChunks,
+  mergeGYTransactionMetadata,
+  makeApiTransactionMetadata,
+  metadataFromApi,
+  metadataToApi,
+  metadataMsg,
+  metadataMsgs
+) where
+
+import qualified Cardano.Api as Api
+import qualified Data.Map as Map
+import           Data.ByteString     (ByteString)
+import           Data.Word           (Word64)
+import           GeniusYield.Imports (Text, coerce)
+
+
+
+newtype GYTxMetadata = GYTxMetadata (Api.TxMetadataInEra Api.BabbageEra)
```

Comment:
Wouldn't it be appropriate if underlying type is `Api.TxMetadata`? Later in tx build phase (`Transaction.hs` file), depending upon whether `gytxMetadata` is `Nothing` or not, corresponding constructor of `TxMetadataInEra` can be used.

Reply:
I think you're right,  the (TxMetadataInEra BabbageEra) structure is only needed at tx build time (e.g. TxMetadataNone when there is no metadata in the tx).  I'll implement the changes.

But then, what is the purpose of having a wrapper `GYTxMetadata` for `Api.TxMetadata`?  (I am guessing that only for consistency of notation and avoiding importing Cardano.Api in code that common users are likely to write.)

Reply:
Because we want users to work with our API and not requiring to bother with internals. If you use Lucid, do you also bother with CML? What if Lucid decides to later move to CSL or some other fork? That will make end user's code invalid if they were interacting directly with internals.

Reply:
Done.